File: Euler/C/p47.c
#include "apremier.h"

int *divPremier(int n, int *length)
{
    if (isprime(n))
    {
        int *divisors = malloc(sizeof(int));
        divisors[0] = n;
        *length = 1;
        return divisors;
    }

    int *divisors = malloc(n * sizeof(int));
    int count = 0;
    for (int i = 1; i < n; i++)
    {
        if (isprime(i) && n % i == 0)
        {
            divisors[count] = i;
            count++;
        }
    }

    int *result = realloc(divisors, count * sizeof(int));
    *length = count;
    return result;
}

int solve()
{
    int max_value = 1000000;
    int i = 1;
    int n = 4;
    while (i < max_value)
    {
        bool allDivisorsN = true;
        for (int j = 0; j < n; j++)
        {
            int length;
            int *divisors = divPremier(i + j, &length);
            if (length != n)
            {
                allDivisorsN = false;
                break;
            }
            free(divisors);
        }

        if (allDivisorsN)
        {
            return i;
        }
        i++;
    }

    return -1;
}

int main()
{
    printf("%d\n", solve());
    return 0;
}

File: Euler/Python/p47.py
from adivisors import divPremier

def solve():
    max_value = 1000000
    i = 30000
    n = 4
    while i < max_value:
        if all(len(divPremier(i + j)) == n for j in range(n)):
            return i
        i += 1
    return -1


if __name__ == "__main__":
    # print(solve())
    print(divPremier(134043), divPremier(134044),divPremier(134045))
